src/lib/models/useModelGist.ts

```diff
@@ -0,0 +1,160 @@
+type GistParams = Omit<BaseGistParams, 'fields' | 'filter'> & {
+    fields?: string | string[]
+    filter?: string | string[]
+}
+
+type GistResourceString = `${string}/gist`
+type ResourceQuery = Query[number]
+type GistResourceQuery = Omit<ResourceQuery, 'resource'> & {
+    resource: GistResourceString
+}
+
+// note that we do not support parallel queries
+// this makes it significantly easier to implement
+// and parallel use of gist queries shouldn't be a common use case
+type GistQuery = {
+    result: GistResourceQuery
+}
+
+type GistQueryResult<Response> = {
+    result: Response
+}
+
+function createGistQuery(
+    resource: GistResourceString,
+    params?: GistParams
+): GistQuery {
+    return {
+        result: {
+            resource: `${resource}`,
+            params: ({ page, ...dynamicParams }) => ({
+                ...params,
+                ...dynamicParams,
+                page,
+            }),
+        },
+    }
+}
+
+function usePagination(
+    refetch: QueryResponse['refetch'],
+    data?: unknown
+): GistPaginator | null {
+    let pager: GistPager | undefined
+
+    if (isDataCollection(data)) {
+        pager = data.pager
+    }
+    const getNextPage = useCallback(() => {
+        if (!pager?.nextPage) {
+            return false
+        }
+        refetch({ page: pager.page + 1 })
+        return true
+    }, [refetch, pager])
+
+    const getPrevPage = useCallback(() => {
+        if (!pager?.prevPage) {
+            return false
+        }
+        refetch({ page: pager.page - 1 })
+        return true
+    }, [refetch, pager])
+
+    return {
+        getNextPage,
+        getPrevPage,
+    }
+}
+
+type GistPaginator = {
+    getNextPage: () => boolean
+    getPrevPage: () => boolean
+}
+
+type BaseUseModelGistResult<Response> = Pick<
+    QueryResponse,
+    'loading' | 'error' | 'called' | 'refetch'
+> & {
+    data?: Response
+}
+
+type UseModelGistResultPaginated<Response> =
+    BaseUseModelGistResult<Response> & {
+        pagination: GistPaginator
+    }
+type UseModelGistResult<Response extends GistResponse> =
+    | BaseUseModelGistResult<Response>
+    | UseModelGistResultPaginated<Response>
+
+const isDataCollection = (
+    data: unknown
+): data is GistCollectionResponse<IdentifiableObject, GistResourceString> => {
+    // gist endpoints are always paged if they're collections
+    return (data as GistCollectionResponse)?.pager !== undefined
+}
+
+export function useModelGist<Response extends GistPagedResponse>(
+    gistResource: GistResourceString,
+    params?: GistParams
+): UseModelGistResultPaginated<Response>
+
+export function useModelGist<Response extends GistObjectResponse>(
+    gistResource: GistResourceString,
+    params?: GistParams
+): UseModelGistResult<Response>
+
+export function useModelGist<Response extends GistResponse>(
+    gistResource: GistResourceString,
+    params?: GistParams
+): UseModelGistResult<Response> {
```

Comment:
This API might be controversial, since I'm moving away from `query`-objects. But you all probably know my feeling towards query-objects, and the quirks that come with them...

The idea of this hook API (passing in a gist-string), is that the `gist`-api returns endpoints as strings to get more information about referenced objects. So you could eg. make a request to `dataElements/id`, and get the endpoints for `dataElementGroups`. You would then use this resulting string (eg. `/dataElements/FTRrcoaog83/dataElementGroups/gist`), and pass that into this hook.

There are some problems with that currently though, since `useDataQuery` doesn't easily support dependent queries. You could use `lazy` - but since the `resource` itself is unknown, and depends on a previous query - you would need to condinally render a component with the hook. We could potentially work around this with a static `/` as the resource, and the dependent part in the `id` of the query. 

Something like this:

```js
result: {
            resource: `/`,
            id: (actualResource) => actualResource, 
            params: ({ page, ...dynamicParams }) => ({
                ...params,
                ...dynamicParams,
                page,
            }),
        },
```